src/script/script_config.cpp

```diff
@@ -129,11 +129,13 @@ void ScriptConfig::ResetSettings()
 	this->settings.clear();
 }
 
-void ScriptConfig::AddRandomDeviation()
+void ScriptConfig::AddRandomDeviation(bool is_ai)
 {
 	for (ScriptConfigItemList::const_iterator it = this->GetConfigList()->begin(); it != this->GetConfigList()->end(); it++) {
 		if ((*it).random_deviation != 0) {
-			this->SetSetting((*it).name, InteractiveRandomRange((*it).random_deviation * 2) - (*it).random_deviation + this->GetSetting((*it).name));
+			int value = this->GetSetting((*it).name);
+			int r = InteractiveRandomRange((*it).random_deviation * 2) - (*it).random_deviation + value;
+			this->SetSetting((*it).name, is_ai && StrEmpty((*it).description) && strcmp((*it).name, "start_date") ? r : (value != 0) ? max(1, r) : 0);
```

Comment:
Seems this is the only place where is_ai is used, is there any way to avoid having to touch so much other stuff just to pass it here?

Reply:
This condition may be sufficient: typeid(this) == typeid(AIConfig *)
You'd need to #incude <typeinfo>, but then you could drop all of the is_ai cruft then.

Reply:
Come to think of it, why disallow 0 here? If it's an AI then start_date 0 is fine, if it's a GS then start_date is... not relevant.